### PR TITLE
Show popup when booking an appointment for the same day

### PR DESCRIPTION
**Pull request by yaron1m**
image

### CODE REVIEW

File: packages/donor/src/components/basic/Popup/Popup.tsx [hunk 1]
```diff
@@ -50,7 +52,10 @@ export default function Popup({
         {image && <img src={image} alt={"popup"} className={styles.image} />}
         <div className={styles.popupText}>
           <div className={styles.popupTextTitle}>{title}</div>
-          <div className={styles.popupTextContent}>{content}</div>
+          <div className={styles.popupTextContent}>
+            {content}
+            {children}
+          </div>
```
Comment (yedidya03) [inline packages/donor/src/components/basic/Popup/Popup.tsx:58]: you can remove the content prop if you use children

File: packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const donationStartTime = new Date(bookedAppointment.donationStartTimeMillis);
+  const bookingTime = new Date(bookedAppointment.bookingTimeMillis);
+  if (!DateUtils.isSameDay(now, donationStartTime)) {
+    return false;
+  }
+
+  // If appointment was booked more than 10s ago
+  if (now.getTime() - bookingTime.getTime() > 10_000) {
+    return false;
+  }
```
Comment (yedidya03) [inline packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts:16]: what is that for?

Comment (yaron1m) [reply packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts:16]: Without this, a user could schedule on Sunday an appointment starting on Monday, and then re-open the app on Monday and see the popup.

Here I verify that the appointment was booked very recently

File: packages/donor/src/components/basic/Popup/Popup.tsx (hunk 1 above)
Comment (yaron1m) [reply packages/donor/src/components/basic/Popup/Popup.tsx:58]: Done

File: packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts (hunk 2 above)
Comment (yaron1m) [reply packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts:16]: Added a comment

Comment (yedidya03) [reply packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts]: but this code is executed only if the appointment is in the same day, so in this usecase the code will not get there at all.

File: packages/donor/src/components/basic/Popup/Popup.tsx [hunk 3]
```diff
@@ -44,13 +46,17 @@ export default function Popup({
     onClose = onBack;
   }
 
+  if (content) {
+    children = content;
+  }
```
Comment (yedidya03) [inline packages/donor/src/components/basic/Popup/Popup.tsx:51]: why do you need content at all?

Comment (yaron1m) [reply packages/donor/src/components/basic/Popup/Popup.tsx]: Done

**Review by yedidya03 — APPROVED** [review]
Nice Work!

**Pull request merged**